src/core/qgsnewsfeedparser.h

```diff
@@ -0,0 +1,169 @@
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSNEWSFEEDPARSER_H
+#define QGSNEWSFEEDPARSER_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include <QObject>
+#include <QUrl>
+#include <QPixmap>
+
+class QgsNetworkContentFetcher;
+
+/**
+ * \ingroup core
+ * Parser for published QGIS news feeds.
+ *
+ * This class is designed to work with the specialized QGIS news feed API. See
+ * https://github.com/elpaso/qgis-feed.
+ *
+ * \since QGIS 3.10
+ */
+class CORE_EXPORT QgsNewsFeedParser : public QObject
+{
+    Q_OBJECT
+  public:
+
+    /**
+     * Represents a single entry from a news feed.
+     */
+    class Entry
+    {
+      public:
+
+        //! Unique entry identifier
+        int key = 0;
+
+        //! Entry title
+        QString title;
+
+        //! Optional URL for image associated with entry
+        QString imageUrl;
+
+        //! Optional image data
+        QPixmap image;
+
+        //! HTML content of news entry
+        QString content;
+
+        //! Optional URL link for entry
+        QUrl link;
+
+        //! TRUE if entry is "sticky" and should always be shown at the top
+        bool sticky = false;
+    };
+
+    /**
+     * Constructor for QgsNewsFeedParser, parsing the specified \a feedUrl.
+     *
+     * The optional \a authcfg argument can be used to specify an authentication
+     * configuration to use when connecting to the feed.
+     */
+    QgsNewsFeedParser( const QUrl &feedUrl, const QString &authcfg = QString(), QObject *parent SIP_TRANSFERTHIS = nullptr );
+
+    /**
+     * Returns a list of existing entries in the feed.
+     */
+    QList< QgsNewsFeedParser::Entry > entries() const;
+
+    /**
+     * Dismisses an entry with matching \a key.
+     *
+     * This removes the entry from the local store, ensuring it will never be present again.
+     *
+     * \see dismissAll()
+     */
+    void dismissEntry( int key );
+
+    /**
+     * Dismisses all current news items.
+     * \see dismissEntry()
+     */
+    void dismissAll();
+
+    /**
+     * Returns the authentication configuration for the parser.
+     */
+    QString authcfg() const;
+
+    /**
+     * Returns the settings key used for a feed with the given \a baseUrl.
+     */
+    static QString keyForFeed( const QString &baseUrl );
+
+  public slots:
+
+    /**
+     * Fetches new entries from the feed's URL.
+     * \see fetched()
+     */
+    void fetch();
```

Comment:
Any reason to make this a slot?

Reply:
Just kinda seems "slotty" to me - e.g. fetching after a timeout, fetching after a button press, ... (I've kept this particular class generic, in case anyone wants to re-use outside of the welcome page)